Upgrade net8

eventsourcer opened the pull request:
Upgrade solutions to net8 adding primary constructors, minimal Apis, immutability design and collection expressions.

eventsourcer committed:
Upgrade to net8

eventsourcer committed:
Add primary constructors, minimal Apis and collection expressions

eventsourcer committed:
Enforce immutability

yvesgoeleven commented on 05 aspnet/OrderBooking.WebAPI/Controllers/CommandController.cs:
```diff
@@ -5,14 +5,9 @@ namespace OrderBooking.WebAPI.Controllers
     [ApiController]
     [Route("api/orderbooking")]
-    public class CommandController : ControllerBase
+    public class CommandController(IEventSourcedRepository<OrderBooking> repo) : ControllerBase
```
The equivalent in lesson 11 has now become a minimal api, this should probably stay consistent across all lessons.

yvesgoeleven commented on 05 aspnet/OrderBooking.WebAPI/Controllers/QueryController.cs:
```diff
@@ -6,14 +6,9 @@ namespace OrderBooking.WebAPI.Controllers
     [Route("api/orderbooking")]
     [ApiController]
-    public class QueryController : ControllerBase
+    public class QueryController(IRestoreProjections<Booking> proj) : ControllerBase
```
The equivalent in lesson 11 has now become a minimal api, this should probably stay consistent across all lessons.

yvesgoeleven commented on 06 task oriented ui/OrderBooking.WebAPI/Controllers/CommandController.cs:
```diff
@@ -5,14 +5,9 @@ namespace OrderBooking.WebAPI.Controllers
     [ApiController]
     [Route("api/orderbooking")]
-    public class CommandController : ControllerBase
+    public class CommandController(IEventSourcedRepository<OrderBooking> repo) : ControllerBase
```
The equivalent in lesson 11 has now become a minimal api, this should probably stay consistent across all lessons.

yvesgoeleven commented on 06 task oriented ui/OrderBooking.WebAPI/Controllers/QueryController.cs:
```diff
@@ -6,14 +6,9 @@ namespace OrderBooking.WebAPI.Controllers
     [Route("api/orderbooking")]
     [ApiController]
-    public class QueryController : ControllerBase
+    public class QueryController(IRestoreProjections<Booking> proj) : ControllerBase
```
The equivalent in lesson 11 has now become a minimal api, this should probably stay consistent across all lessons.

yvesgoeleven commented on 07 transient reaction/OrderBooking.WebAPI/Controllers/CommandController.cs:
```diff
@@ -5,14 +5,9 @@ namespace OrderBooking.WebAPI.Controllers
     [ApiController]
     [Route("api/orderbooking")]
-    public class CommandController : ControllerBase
+    public class CommandController(IEventSourcedRepository<OrderBooking> repo) : ControllerBase
```
The equivalent in lesson 11 has now become a minimal api, this should probably stay consistent across all lessons.

yvesgoeleven commented on 07 transient reaction/OrderBooking.WebAPI/Controllers/QueryController.cs:
```diff
@@ -6,9 +6,9 @@ namespace OrderBooking.WebAPI.Controllers
     [Route("api/orderbooking")]
     [ApiController]
-    public class QueryController : ControllerBase
+    public class QueryController(IRestoreProjections<Booking> proj) : ControllerBase
```
The equivalent in lesson 11 has now become a minimal api, this should probably stay consistent across all lessons.

yvesgoeleven commented on 08 guaranteed delivery/OrderBooking.WebAPI/Controllers/CommandController.cs:
```diff
@@ -5,14 +5,9 @@ namespace OrderBooking.WebAPI.Controllers
     [ApiController]
     [Route("api/orderbooking")]
-    public class CommandController : ControllerBase
+    public class CommandController(IEventSourcedRepository<OrderBooking> repo) : ControllerBase
```
The equivalent in lesson 11 has now become a minimal api, this should probably stay consistent across all lessons.

yvesgoeleven commented on 08 guaranteed delivery/OrderBooking.WebAPI/Controllers/QueryController.cs:
```diff
@@ -6,14 +6,9 @@ namespace OrderBooking.WebAPI.Controllers
     [Route("api/orderbooking")]
     [ApiController]
-    public class QueryController : ControllerBase
+    public class QueryController(IRestoreProjections<Booking> proj) : ControllerBase
```
The equivalent in lesson 11 has now become a minimal api, this should probably stay consistent across all lessons.

yvesgoeleven commented on 09 polyglot persistence/OrderBooking.WebAPI/Controllers/CommandController.cs:
```diff
@@ -5,14 +5,9 @@ namespace OrderBooking.WebAPI.Controllers
     [ApiController]
     [Route("api/orderbooking")]
-    public class CommandController : ControllerBase
+    public class CommandController(IEventSourcedRepository<OrderBooking> repo) : ControllerBase
```
The equivalent in lesson 11 has now become a minimal api, this should probably stay consistent across all lessons.

yvesgoeleven commented on 09 polyglot persistence/OrderBooking.WebAPI/Controllers/QueryController.cs:
```diff
@@ -6,14 +6,9 @@ namespace OrderBooking.WebAPI.Controllers
     [Route("api/orderbooking")]
     [ApiController]
-    public class QueryController : ControllerBase
+    public class QueryController(IRestoreProjections<Booking> proj) : ControllerBase
```
The equivalent in lesson 11 has now become a minimal api, this should probably stay consistent across all lessons.

yvesgoeleven commented on 10 manual task processing/OrderBooking.WebAPI/Controllers/CommandController.cs:
```diff
@@ -5,14 +5,9 @@ namespace OrderBooking.WebAPI.Controllers
     [ApiController]
     [Route("api/orderbooking")]
-    public class CommandController : ControllerBase
+    public class CommandController(IEventSourcedRepository<OrderBooking> repo) : ControllerBase
```
The equivalent in lesson 11 has now become a minimal api, this should probably stay consistent across all lessons.

yvesgoeleven commented on 10 manual task processing/OrderBooking.WebAPI/Controllers/QueryController.cs:
```diff
@@ -7,16 +7,10 @@ namespace OrderBooking.WebAPI.Controllers
     [Route("api/orderbooking")]
     [ApiController]
-    public class QueryController : ControllerBase
+    public class QueryController(IRestoreProjections<Booking> proj, SearchClient search) : ControllerBase
```
The equivalent in lesson 11 has now become a minimal api, this should probably stay consistent across all lessons.

eventsourcer replied:
Applied minimal Apis to all solutions.

yvesgoeleven commented on 11 automated task processing/OrderBooking.ComponentTests/WhenReactingToBookingStarted.cs:
```diff
@@ -14,22 +14,22 @@ namespace OrderBooking.ComponentTests
+        private readonly Mock<IHubClients> mockHubClients = new();
+        private readonly Mock<IHubContext<EventsHub>> mockHubContext = new();
+        private readonly Mock<IEmailService> mockEmailService = new();
+        public ReactingToBookingStarted()
```
Here the constructor name is incorrect, these setups won't be called.

eventsourcer replied:
Fixed all test errors.

yvesgoeleven commented on 11 automated task processing/OrderBooking.WebAPI/Program.cs:
```diff
@@ -91,8 +98,45 @@
+
+    return Results.Ok(aggregate.Id);
+});
+
```
I would personally expose these as a UseOrderBooking() and UseNotificationPreferences() extension methods on the app.

eventsourcer replied:
Great idea, this is what I initially wanted to do but kept it simple. added two extensions for both aggregates please check it out.

eventsourcer committed:
Create extensions for minimal Apis

eventsourcer committed:
Apply minimal Apis to remaining solutions

eventsourcer committed:
Fix unit and integration tests

yvesgoeleven commented on 11 automated task processing/OrderBooking.WebAPI/MinimalApiConfig.cs:
```diff
@@ -0,0 +1,78 @@
+using FastExpressionCompiler;
+using MessageHandler.EventSourcing.DomainModel;
+using Microsoft.Extensions.Azure;
+using OrderAggregate = OrderBooking.OrderBooking;
```
From a business analyses perspective I really don't like this rename. The aggregate is representing the Order Booking business capability. It's not an 'Order', which is something that doesn't exist, this is a step that transitions from PurchaseOrder to SalesOrder.

If you want to rename it to avoid the Namespace.Class issue, than I'd suggest to rename it to 'Booking'

eventsourcer replied:
Entirely agree, yes the reason of renaming is because of VS Code having worried about class and namespace being identical.

yvesgoeleven commented on 11 automated task processing/OrderBooking.WebAPI/MinimalApiConfig.cs:
```diff
@@ -0,0 +1,78 @@
+using MessageHandler.EventSourcing.DomainModel;
+using Microsoft.Extensions.Azure;
+using OrderAggregate = OrderBooking.OrderBooking;
+using NotificationAggregate = NotificationPreferences.NotificationPreferences;
```
Similar to the above, either keep the name as is, or rename it to 'Preferences'

eventsourcer replied:
Preferences and Booking, will follow these conventions on all updated solutions.

yvesgoeleven commented on 11 automated task processing/OrderBooking.WebAPI/Program.cs:
```diff
@@ -1,11 +1,18 @@
+using OrderBooking.WebAPI.Controllers;
 using OrderBooking.WebAPI.SignalR;
+using OrderAggregate = OrderBooking.OrderBooking;
+using NotificationAggregate = NotificationPreferences.NotificationPreferences;
```
The two lines above can be removed, not used anymore

eventsourcer replied:
What about MinimalApiConfig, any ideas of renaming it?

yvesgoeleven replied:
Perhaps call it ApiRoutingConfiguration?

eventsourcer replied:
Sounds great

yvesgoeleven commented:
Thanks @eventsourcer 

I will merge it in this weekend, and update the corresponding text where needed. Looks like only 3 pages need an update.

eventsourcer commented:
My pleasure.
Sure, you can consider only primary changes such as minimal apis, and classes being made immutable.

eventsourcer committed:
Following better naming conventions

eventsourcer commented:
Added one more commit, adhering to proposed naming conventions.

yvesgoeleven commented:
@eventsourcer I've now updated all the texts as well.

Mind you, I had to make one more change, the /confirm route in lesson 11 accidentally got changed from PUT to a POST, I reverted that back to a PUT.

eventsourcer commented:
> @eventsourcer I've now updated all the texts as well.
> 
> Mind you, I had to make one more change, the /confirm route in lesson 11 accidentally got changed from PUT to a POST, I reverted that back to a PUT.

Thanks a lot @yvesgoeleven for updating learning path text.
ah true missed that one :), thanks for fixing it.
